train/NEAT/evolve.py

```diff
@@ -0,0 +1,157 @@
+    #for x in range(args.generations if args.generations > 0 else sys.maxsize): 
+    while True:
+        winner = pop.run(pe.evaluate, 5)
+        
+        if args.vineLogging:
+            #TODO: Fix Vine Logging, everything needed should be in stats. 
+            visualise.save_statistics(stats)
+        else:
+            visualise.plot_stats(stats, ylog=False, view=False)
+            visualise.plot_species(stats, view=False)
+
+        # Save the best Genome from the last 5 gens.
+        with open('Best-{}.pkl'.format(len(stats.most_fit_genomes)), 'wb') as output:
+            pickle.dump(winner, output, 1)
+
+def playBest(config, bestFile):
+    #TODO: Set to normal...
+    # Error - The mode "normal" is not supported. Supported options are "algo" or "human"
+    wrapper = SetPlayingMode('human')
+    e = wrapper(env[0])
+
+    genome = pickle.load(open(bestFile, 'rb'))
+
+    net = neat.nn.FeedForwardNetwork.create(genome, config)
+
+    done = False
+    observation = e.reset()
+    while not done:
+        action = clean_outputs(net.activate(observation.flatten()))
+        
+        print('Action taken by NN : [{} {} {} {} {} {}]'.format(
+                ('UP' if action[0] else ''),
+                ('Left' if action[1] else ''),
+                ('Right' if action[2] else ''),
+                ('Down' if action[3] else ''),
+                ('A' if action[4] else ''),
+                ('B' if action[5] else '')))
+                
+        e.step(action)
+
+def load_config(config_file):
+    local_dir = os.path.dirname(__file__)
+    config_path = os.path.join(local_dir, config_file)
+
+    config = neat.Config(neat.DefaultGenome, neat.DefaultReproduction,
+                         neat.DefaultSpeciesSet, neat.DefaultStagnation,
+                         config_path)
+    return config
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Mario NEAT Trainer')
+    parser.add_argument('--config-file', dest="configFile", type=str, default="config-feedforward",
+                        help="The path to the NEAT parameter config file to use")
+    parser.add_argument('--generations', dest="generations", type=int, default=-1,
+                        help="The number of generations to evolve the network")
+    parser.add_argument('--play-best', dest="playBest", type=str, default="",
+                        help="Play the best of a trained network")
+    parser.add_argument('--checkpoint', dest="checkpoint", type=str, default="",
+                        help="Resmume training from a saved checkpoint")
+    parser.add_argument('--num-cores', dest="numCores", type=int, default=4,
+                        help="The number of cores on your computer for parallel execution")
+    parser.add_argument('--vine-logging', dest="vineLogging", action='store_true',
+                        help="Use Vine Logging")
+    args = parser.parse_args()
+
+    config = load_config(args.configFile)
```

Comment:
We are missing a display bind like we had - is this just not needed anymore for Wooey?

Reply:
In it current form it wont work with wooey because it will never stop running. but as nice as wooey is, do we need it?

Reply:
Actually either way it is good to have it there. ill put it back in